src/gi.cc

```diff
@@ -303,6 +304,27 @@ NAN_METHOD(StructFieldGetter) {
     bool mustCopy = true;
     BaseInfo typeInfo = g_field_info_get_type(*fieldInfo);
 
+    long length = -1;
+    if (jsLengthFieldInfo->IsObject()) {
+        if (jsLengthFieldInfo->InternalFieldCount() == 0) {
+            Nan::ThrowError("StructFieldGetter: length field info is invalid");
+            return;
+        }
+        BaseInfo lengthFieldInfo =
+            g_base_info_ref((GIFieldInfo *) GNodeJS::PointerFromWrapper(jsLengthFieldInfo));
+        if (jsLengthFieldInfo.IsEmpty()) {
```

Comment:
Shouldn't this check come before `PointerFromWrapper()`? And can this check really fail if the two previous ones (`IsObject` and `InternalFieldCount`) have passed?

Reply:
you're right, this is a typo. I was replicating the existing checks and this was supposed to say `lengthFieldInfo.isEmpty()`